app/components/profile_picture_button.js

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+
+import {Client4} from 'mattermost-redux/client';
+
+import {t} from 'app/utils/i18n';
+
+import AttachmentButton from 'app/components/attachment_button';
+
+export default class ProfilePictureButton extends PureComponent {
+    static propTypes = {
+        currentUser: PropTypes.object.isRequired,
+        theme: PropTypes.object.isRequired,
+        extraOptions: PropTypes.func,
+    };
+
+    state = {
+        profileImageUri: null,
+    };
+
+    getProfileImageUri = async (id, lastPictureUpdate) => {
+        let uri;
+        try {
+            uri = await Client4.getProfilePictureUrl(id, lastPictureUpdate);
+        } catch (error) {
+            return null;
+        }
+        return this.setState({profileImageUri: uri});
+    }
+
+    isCustomProfileImage = async (id, lastPictureUpdate) => {
+        let uri;
+        let action = null;
+        try {
+            uri = await Client4.getProfilePictureUrl(id, lastPictureUpdate);
```

Comment:
`Client4.getProfilePictureUrl` is actually synchronous and won't throw errors, so we don't need the try/catch or await here